src/main/java/core/basesyntax/strategy/handler/DecrementHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.OperationHandler;
+
+public class DecrementHandler implements OperationHandler {
+    @Override
+    public void apply(Fruit fruit, int quantity) {
+        Fruit existingFruit = Storage.getFruit(fruit.getName());
+        if (existingFruit == null || existingFruit.getQuantity() < quantity) {
```

Comment:
The check for `existingFruit.getQuantity() < quantity` should ensure the result balance is positive as per the checklist requirement 'Handling Purchase operation'. You should also consider the case where `existingFruit.getQuantity()` is equal to `quantity`.